test/performance/odp_ipsecfwd.c

```diff
@@ -1710,28 +1731,18 @@ static odp_bool_t setup_pktios(prog_config_t *config)
 	return true;
 }
 
-static odp_bool_t setup_fwd_table(prog_config_t *config)
+static int compare_fwd_entries(const void *first, const void *second)
 {
-	fwd_entry_t *fwd_e;
-
-	config->fwd_tbl = odph_iplookup_table_create(SHORT_PROG_NAME "_fwd_tbl", 0U, 0U,
-						     sizeof(fwd_entry_t *));
+	const fwd_entry_t *entr1 = first;
+	const fwd_entry_t *entr2 = second;
+	const int64_t val = (int64_t)entr2->prefix - entr1->prefix;
 
-	if (config->fwd_tbl == NULL) {
-		ODPH_ERR("Error creating forwarding table\n");
-		return false;
-	}
-
-	for (uint32_t i = 0U; i < config->num_fwds; ++i) {
-		fwd_e = &config->fwd_entries[i];
-
-		if (odph_iplookup_table_put_value(config->fwd_tbl, &fwd_e->prefix, &fwd_e) < 0) {
-			ODPH_ERR("Error populating forwarding table\n");
-			return false;
-		}
-	}
+	return val < 0 ? -1 : val == 0 ? 0 : 1;
```

Comment:
I wonder what is the rationale for sorting the entries by prefix. To get more repeatable test results regarless of the order of the entries in the config file?

It would be good to sort first by mask length so that longest prefices appear first in the lookup array. This way the expected longest-prefix match works. The prefix value could be a secondary sort key if you think it is useful.

Reply:
> longest prefices appear first

They do now, after the sorting, which was the idea.

Reply:
No, longest prefixes do not appear first. The comparison function of the sort is not even looking at prefix length.

Reply:
Not sure if we're talking about the same thing.

E.g. passing:
```
fwd: (
        {
        prefix: "192.168.134.0/0"
        if: "ens9f0"
        dst_mac: "00:00:05:00:07:00"
        },
	{
	prefix: "192.168.1.0/24"
	if: "ens9f0"
	dst_mac: "00:00:05:00:07:00"
	},
	{
	prefix: "192.168.1.11/32"
	if: "ens9f0"
	dst_mac: "00:00:05:00:07:00"
	},
	{
	prefix: "192.168.134.0/16"
	if: "ens9f0"
	dst_mac: "00:00:05:00:07:00"
	},
	{
	prefix: "192.168.134.0/8"
	if: "ens9f0"
	dst_mac: "00:00:05:00:07:00"
	}
);
```
will result in this order in the table:
```
192.168.1.11
192.168.1.0
192.168.0.0
192.0.0.0
0.0.0.0
```
which is probably enough.

Reply:
Think about e.g. 10.0.0.0/8 and 10.0.0.0/16. Or add 192.168.1.0/25 in your list of routes.

Reply:
Removed sorting in v3, entries are looked up in the config order.